modules/nw-running-dpdk-rootless-tap.adoc

```diff
@@ -0,0 +1,269 @@
+[id="nw-running-dpdk-rootless-tap_{context}"]
+= Using the TAP CNI to run a rootless DPDK workload with kernel access
+
+DPDK applications can use `virtio-user` as an exception path to inject certain types of packets for example log messages into the kernel for processing. For more information about this feature see link:https://doc.dpdk.org/guides/howto/virtio_user_as_exception_path.html[Virtio_user as Exception Path].
+
+In OpenShift Container Platform version 4.14 and later, non-privileged pods can be used to run DPDK applications alongside the tap CNI plugin. The pod also needs the `vhost-net` device to be mounted which can be done by enabling the `needVhostNet` parameter of the SriovNetworkNodePolicy object.
+
+.DPDK and TAP example configuration
+image::rootless_dpdk_example.png[DPDK and TAP plugin]
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Install the SR-IOV Network Operator.
+* Log in as a user with `cluster-admin` privileges.
+* Ensure `setsebools container_use_devices=on` as root on all nodes.
++
+[NOTE]
+====
+Use the Machine Config Operator to set this SELinux boolean.
+====
+
+.Procedure
+
+. Create a dedicated container namespace where you want to deploy your pod by using the following command:
++
+[source,terminal]
+----
+$ oc new-project test-namespace
+----
+
+. Create the following `SriovNetworkNodePolicy` object, and then save the YAML in the `sriov-node-network-policy.yaml` file:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+ name: sriovnic
+ namespace: openshift-sriov-network-operator
+spec:
+ deviceType: netdevice <1>
+ isRdma: true
+ needVhostNet: true
+ nicSelector:
+   vendor: "15b3" <2>
+   deviceID: "101b" <3>
+   rootDevices: ["00:05.0"]
+ numVfs: 10
+ priority: 99
+ resourceName: sriovnic
+ nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+----
+<1> This indicates that the profile is tailored specifically for Mellanox Network Interface Controllers (NICs).
+<2> The vendor hexadecimal code of the SR-IOV network device. The value 15b3 is associated with a Mellanox NIC.
+<3> The device hexadecimal code of the SR-IOV network device.
+
+. Create the `SriovNetworkNodePolicy` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f sriov-node-network-policy.yaml
+----
++
+[NOTE]
+=====
+See the `Configuring SR-IOV network devices` section for a detailed explanation on each option in `SriovNetworkNodePolicy`.
+
+When applying the configuration specified in a `SriovNetworkNodePolicy` object, the SR-IOV Operator may drain the nodes, and in some cases, reboot nodes.
+It may take several minutes for a configuration change to apply.
+Ensure that there are enough available nodes in your cluster to handle the evicted workload beforehand.
+
+After the configuration update is applied, all the pods in `openshift-sriov-network-operator` namespace will change to a `Running` status.
```

Comment:
May be we need to add that mcp gets updated and we need to wait until all nodes are moved to **READYMACHINECOUNT** and UPDATED==true

NAME        CONFIG                                                UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
master      rendered-master-e5e0c8e8be9194e7c5a882e047379cfa      True      False      False      3              3                   3                     0                      7d6h
worker   rendered-workercnf-3b87971b3d368847bff7ff11930006ea   True     False      False      3              3                   3                     0                      7d4h